app/src/main/java/com/example/movieapp/ui/viewmodel/DetailViewModel.kt

```diff
@@ -0,0 +1,75 @@
+package com.example.movieapp.ui.viewmodel
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MediatorLiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Transformations
+import com.example.movieapp.App
+import com.example.movieapp.data.entities.ApiEmptyResponse
+import com.example.movieapp.data.entities.ApiErrorResponse
+import com.example.movieapp.data.entities.ApiSuccessResponse
+import com.example.movieapp.data.model.Movie
+import com.example.movieapp.ui.fragment.DetailFragment
+
+private const val TAG = "DetailViewModel"
+
+class DetailViewModel : AbstractViewModel(), DetailFragment.DetailFragmentListener {
+
+    private var currentId: MutableLiveData<String?> = MutableLiveData()
+    private var currentMoviePair = MediatorLiveData<Pair<Movie?, Boolean>>()
+    private var movieCurrent = Transformations.switchMap(currentId) {
+        it?.let { internetCall(it) } ?: null
+    }
+    private var isLikedMovie: LiveData<Boolean> =
+        Transformations.switchMap(currentId) {
+            Log.i(TAG, "getMovieAndIsLiked: ${it}")
+            it?.toInt()?.let { it1 -> App.database.movieDAO().isLiked(it1) }
+                ?: MutableLiveData()
+        }
+
+    init {
+        currentMoviePair.addSource(movieCurrent) {
+            currentMoviePair.value = Pair(it, isLikedMovie?.value ?: false)
+        }
+        currentMoviePair.addSource(isLikedMovie) { isLiked ->
+            Log.i(TAG, "isLiked : modif de la list de movie ")
+            currentMoviePair.value =
+                Pair(movieCurrent.value, isLiked)
+        }
+    }
+
+    fun getMovieAndIsLiked(idMovie: String): LiveData<Pair<Movie, Boolean>> {
```

Comment:
why not returning a MediatorLiveData<Pair<Movie?, Boolean>> ?

Reply:
Return null 
L'abonnement se fait dans une autre méthode ( _currentMoviePaire)

Reply:
private var _currentMoviePair = MediatorLiveData<Pair<Movie?, Boolean>>()
val currentMoviePair: LiveData<Pair<Movie?, Boolean>>
    get() = _currentMoviePair